File: ABS/8.py
N = int(input())

d = []
for i in range(N):
    d.append(int(input()))

d.sort()
ans = 1
for i in range(len(d)-1, 0, -1):
    if d[i] > d[i-1]: ans += 1
    elif d[i] == d[i-1]: continue
    else: break

print(ans)

File: ABS/8.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N, ans, now;
    cin >> N;
    ans = 1;
    vector<int> D(N);
    for (int i = 0; i < N; i++) cin >> D[i];
    sort(D.begin(), D.end());
    now = D[N-1];
    for (int i = N-2; i>= 0; i--) {
        if (now > D[i]) {
            ans ++;
            now = D[i];
        }
    }
    cout << ans << endl;
}